contracts/toCKB-typescript/src/switch/mint_xt/mod.rs

```diff
@@ -1,7 +1,282 @@
+    let previous_diff = U256::from_little_endian(difficulty.previous().raw_data());
+    let current_diff = U256::from_little_endian(difficulty.current().raw_data());
+    let first_header_diff = headers.index(0).difficulty();
+    debug!(
+        "previous: {:?}, current: {:?}, first_header_diff: {:?}",
+        previous_diff, current_diff, first_header_diff
+    );
+
+    let req_diff = if first_header_diff == current_diff {
+        current_diff
+    } else if first_header_diff == previous_diff {
+        previous_diff
+    } else {
+        return Err(Error::NotAtCurrentOrPreviousDifficulty);
+    };
+
+    if observed_diff < req_diff * TX_PROOF_DIFFICULTY_FACTOR {
+        return Err(Error::InsufficientDifficulty);
+    }
+    debug!("finish diff verify");
+
+    // verify tx
+    let header = headers.index(headers.len() - 1);
+    let mut idx = [0u8; 8];
+    idx.copy_from_slice(proof.index().raw_data());
+    debug!("tx_id: {}", hex::encode(tx_id.as_ref()));
+    debug!("merkle_root: {}", hex::encode(header.tx_root().as_ref()));
+    debug!(
+        "proof: {}",
+        hex::encode(proof.intermediate_nodes().raw_data())
+    );
+    debug!("index: {}", u64::from_le_bytes(idx));
+    if !validatespv::prove(
+        tx_id,
+        header.tx_root(),
+        &MerkleArray::new(proof.intermediate_nodes().raw_data())?,
+        u64::from_le_bytes(idx),
+    ) {
+        return Err(Error::BadMerkleProof);
+    }
+    debug!("finish merkle proof verify");
+
+    Ok(())
+}
+
+fn verify_xt_issue(data: &ToCKBCellDataView) -> Result<(), Error> {
+    match get_xchain_kind()? {
+        XChainKind::Btc => verify_btc_xt_issue(data),
+        XChainKind::Eth => todo!(),
+    }
+}
+
+fn verify_btc_xt_issue(data: &ToCKBCellDataView) -> Result<(), Error> {
+    let lock_hash = load_cell_lock_hash(0, Source::Input)?;
+    debug!("lockscript hash: {:?}", hex::encode(lock_hash));
+    let input_xt_num = QueryIter::new(load_cell_type, Source::Input)
+        .filter(|type_opt| type_opt.is_some())
+        .map(|type_opt| type_opt.unwrap())
+        .filter(|script| {
+            script.code_hash().raw_data().as_ref() == SUDT_CODE_HASH.as_ref()
+                && script.args().raw_data().as_ref() == lock_hash.as_ref()
+        })
+        .count();
+    if input_xt_num != 0 {
+        return Err(Error::InvalidXTInInput);
+    }
+    let xt_amount = data.get_btc_lot_size()?.get_sudt_amount();
+    debug!("xt_amount: {}", xt_amount);
+    let expect = [
+        (
+            1,
+            data.user_lockscript.as_ref(),
+            xt_amount - xt_amount * SIGNER_FEE_RATE.0 / SIGNER_FEE_RATE.1,
+        ),
+        (
+            2,
+            data.signer_lockscript.as_ref(),
+            xt_amount * SIGNER_FEE_RATE.0 / SIGNER_FEE_RATE.1,
+        ),
+    ];
+    debug!("expect: {:?}", expect);
+
+    for (i, lockscript, amout) in expect.iter() {
+        let script = load_cell_type(*i, Source::Output)?;
```

Comment:
maybe add a comment:  fixed order of output cells is required, user-sudt-cell should be outputs[1], signer-sudt-cell should be outputs[2]